drivers/SmartThings/zwave-switch/src/qubino-switches/qubino-relays/qubino-flush-2-relay/init.lua

```diff
@@ -11,43 +11,159 @@
+    device:set_find_child(find_child)
+    device:set_component_to_endpoint_fn(component_to_endpoint)
+  end
+end
+
+local function get_child_metadata(device, endpoint)
+  local name
+  local profile_name
+  if endpoint ~= CHILD_TEMP_SENSOR_EP then
+    name = string.format("Qubino Switch %d", endpoint)
+    profile_name = "metering-switch"
+  else
+    name = "Qubino Temperature Sensor"
+    profile_name = "child-temperature"
+  end
+
+  return {
+    type = "EDGE_CHILD",
+    label = name,
+    profile = profile_name,
+    parent_device_id = device.id,
+    parent_assigned_child_key = string.format("%02X", endpoint),
+    vendor_provided_label = name
+  }
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= st_device.NETWORK_TYPE_CHILD then
+    driver:try_create_device(get_child_metadata(device, CHILD_SWITCH_EP))
+    if device:is_cc_supported(cc.SENSOR_MULTILEVEL) then
+      driver:try_create_device(get_child_metadata(device, CHILD_TEMP_SENSOR_EP))
     end
+  end
 end
 
-local function endpoint_to_component(device, ep)
-    if ep == 2 then
-      return string.format("switch%d", ep)
-    elseif ep == 3 then
-      return "extraTemperatureSensor"
+local function do_refresh(driver, device, cmd)
+  if device:supports_capability(capabilities.switch) then
+    device:send_to_component(SwitchBinary:Get({}), cmd.component)
+  end
+  if device:supports_capability(capabilities.powerMeter) then
+    device:send_to_component(Meter:Get({ scale = Meter.scale.electric_meter.WATTS }), cmd.component)
+  end
+  if device:supports_capability(capabilities.energyMeter) then
+    device:send_to_component(Meter:Get({ scale = Meter.scale.electric_meter.KILOWATT_HOURS }), cmd.component)
+  end
+  if device:supports_capability(capabilities.temperatureMeasurement) then
+    device:send_to_component(SensorMultilevel:Get({ sensor_type = SensorMultilevel.sensor_type.TEMPERATURE }), cmd.component)
+  end
+end
+
+local function switch_report_handler(driver, device, cmd)
+  local event
+  local value = cmd.args.target_value or cmd.args.value
+
+  if value ~= nil then
+    if value == SwitchBinary.value.OFF_DISABLE then
+      event = capabilities.switch.switch.off()
     else
-      return "main"
+      event = capabilities.switch.switch.on()
     end
+  end
+
+  device:emit_event_for_endpoint(cmd.src_channel, event)
 end
 
-local function map_components(self, device)
-    device:set_endpoint_to_component_fn(endpoint_to_component)
-    device:set_component_to_endpoint_fn(component_to_endpoint)
+local function set_switch(value)
+  return function(driver, device, cmd)
+    local delay = constants.DEFAULT_GET_STATUS_DELAY
+    local query_device = function()
+      do_refresh(driver, device, cmd)
+    end
+
+    device:send_to_component(SwitchBinary:Set({ switch_value = value }), cmd.component)
+    device.thread:call_with_delay(delay, query_device)
+  end
+end
+
+local function sensor_multilevel_report(driver, device, cmd)
```

Comment:
I think the DTH also will create the child temperature sensor if it receives a temp event after it's been added.

Reply:
done